src/main/java/ch/njol/skript/conditions/CondIsWhitelisted.java

```diff
@@ -19,72 +19,72 @@
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Condition;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
 @Name("Is Whitelisted")
-@Description("Whether or not the server or a player is whitelisted.")
-@Examples({"if server is whitelisted:", "if player is whitelisted"})
-@Since("2.5.2")
+@Description("Whether or not the server or a player is whitelisted, and server is whitelist enforced.")
+@Examples({
+	"if player is whitelisted:",
+	"if server is whitelisted:",
+	"if server is whitelist enforced:"
+})
+@Since("2.5.2, INSERT VERSION (enforce)")
+@RequiredPlugins("Minecraft 1.17+")
 public class CondIsWhitelisted extends Condition {
-	
+
+	private static final boolean ENFORCE_SUPPORT;
+
 	static {
+		ENFORCE_SUPPORT = Skript.methodExists(Bukkit.class, "isWhitelistEnforced");
 		Skript.registerCondition(CondIsWhitelisted.class,
 			"[the] server (is|1¦is(n't| not)) white[ ]listed",
-			"%players% (is|are)(|1¦(n't| not)) white[ ]listed");
+			"%offlineplayers% (is|are)(|1¦(n't| not)) white[ ]listed",
+			(ENFORCE_SUPPORT ? "[the] server (is|1¦is(n't| not)) white[ ]list enforced" : ""));
 	}
-	
+
 	@Nullable
-	private Expression<Player> player;
-	
+	private Expression<OfflinePlayer> player;
+
 	private boolean isServer;
-	
+	private boolean isEnforce;
+
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setNegated(parseResult.mark == 1);
-		isServer = matchedPattern == 0;
-		if (matchedPattern == 1)
-			player = (Expression<Player>) exprs[0];
+		if (matchedPattern == 0 || matchedPattern == 2) {
+			isServer = true;
+			isEnforce = matchedPattern == 2;
+		}
+		else if (matchedPattern == 1)
+			player = (Expression<OfflinePlayer>) exprs[0];
 		return true;
 	}
-	
+
 	@Override
-	@SuppressWarnings("null")
-	public boolean check(Event e) {
+	public boolean check(Event event) {
 		if (isServer)
-			return Bukkit.hasWhitelist() == isNegated();
-		Player[] players = player.getAll(e);
-		if (player.getAnd() && isNegated()) {
-			for (Player player : players)
-				if (player.isWhitelisted())
-					return false;
-		} else if(player.getAnd()){
-			for (Player player : players)
-				if (!player.isWhitelisted())
-					return false;
-		} else {
-			for (Player player: players)
-				if(player.isWhitelisted())
-					return !isNegated();
-		}
-		return !isNegated();
+			return (isEnforce ? Bukkit.isWhitelistEnforced() : Bukkit.hasWhitelist()) ^ isNegated();
+
+		return player.check(event, OfflinePlayer::isWhitelisted, isNegated());
 	}
-	
+
 	@Override
-	@SuppressWarnings("null")
-	public String toString(@Nullable Event e, boolean debug) {
-		return (player.getSingle(e) != null ? "player" : "server") + (isNegated() ? "not" : "") + "  whitelisted";
+	public String toString(@Nullable Event event, boolean debug) {
+		return (isServer ? "server" : "player") + " is " + (isNegated() ? "not" : "") + " "
```

Comment:
```suggestion
		return (player != null ? player.toString(e, debug) : "server") + " is " + (isNegated() ? "not" : "") + " "
```
Not sure why it wasn't specified like this in the old version 😅